enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,145 @@
+---
+title: olm-integration-with-openshift-monitoring
+authors:
+  - "@awgreene"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-12-11
+last-updated: 2019-12-11
+status: implementable
+---
+
+# olm-integration-with-openshift-monitoring
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary and Motivation
+
+The primary purpose of this enhancement is to propose a series of changes that would enable a set of Red Hat approved operators to report metrics to [Telemeter](https://github.com/openshift/telemeter).
+
+First, [Operator-Lifecycle-Manager (OLM)](https://github.com/operator-framework/operator-lifecycle-manager) managed operators should be capable of defining `ServiceMonitors` and `PrometheusRule` objects to interact with generic [Prometheus Operators](https://github.com/coreos/prometheus-operator) on cluster.
+
+Second, OLM should automatically integrate a subset of approved operators with the [OpenShift Monitoring](https://github.com/openshift/cluster-monitoring-operator) Prometheus Operator present on all OpenShift clusters. At this point, a subset of these operator could work with the Telemeter team to whitelist metrics that should be scraped by Telemeter.
+
+For those unfamiliar with Telemeter, Red Hat uses metrics collected by Telemeter to:
+
+- Proactively reach out to customers with "problem clusters"
+- Troubleshoot customer issues
+- Understand how individual applications are behaving at a holistic level
+
+### Goals
+
+- Provide OLM managed operators with a first-class way of creating `ServiceMonitors` and `PrometheusRules` objects to interact with the Prometheus Operator
+- Automatically integrating Red Hat approved Operators with the OpenShift Monitoring Prometheus Operator, allowing them to work with the Telemeter team to whitelist their metrics.
+
+### Non-Goals
+
+- Provide generic metrics about OLM managed operators
+- Whitelist OLM managed operator metrics so they are picked up by Telemeter.
+
+## Proposal
+
+### Generic Prometheus Operator Support
+
+#### OLM support for new Prometheus objects in bundles
+
+The bundle object currently has the following format:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   └── example.rbac.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+Notice that OLM understands how to create and manage CSVs, CRDs, and RBAC files. This enhancement proposes expanding this feature to support `ServiceMonitors` and `PrometheusRules` objects:
```

Comment:
how does OLM make the decision between "resources that get their own file in the manifests dir" and "resources that are defined in the CSV"?  How was the choice made here?